internal/clientconn/conn.go

```diff
@@ -433,80 +439,141 @@ func (c *conn) route(connCtx context.Context, reqHeader *wire.MsgHeader, reqBody
 
 	c.m.Requests.WithLabelValues(reqHeader.OpCode.String(), command).Inc()
 
-	// set body for error
+	resHeader, err = c.responseHeader(resHeader.OpCode, reqHeader.RequestID, resBody)
 	if err != nil {
-		switch resHeader.OpCode {
-		case wire.OpCodeMsg:
-			protoErr := mongoerrors.Make(connCtx, err, "", c.l)
-			resBody = protoErr.Msg()
-			result = protoErr.Name
-			argument = protoErr.Argument
-
-		case wire.OpCodeReply:
-			protoErr := mongoerrors.Make(connCtx, err, "", c.l)
-			resBody = protoErr.Reply()
-			result = protoErr.Name
-			argument = protoErr.Argument
+		result = ""
 
-		case wire.OpCodeQuery:
-			fallthrough
-		case wire.OpCodeUpdate:
-			fallthrough
-		case wire.OpCodeInsert:
-			fallthrough
-		case wire.OpCodeGetByOID:
-			fallthrough
-		case wire.OpCodeGetMore:
-			fallthrough
-		case wire.OpCodeDelete:
-			fallthrough
-		case wire.OpCodeKillCursors:
-			fallthrough
-		case wire.OpCodeCompressed:
```

Comment:
I don't think this case was ever run, because `resHeader.OpCode` wasn't set in previous switch statement, but I think there was a reason we wanted to log and collect metrics with `unhandled` instead of `unexpected`.

Reply:
That's because of `fallthrough`